codec-http/src/main/java/io/netty/handler/codec/http/HttpMethod.java

```diff
@@ -143,18 +133,14 @@ private HttpMethod(String name, boolean bytes) {
             }
         }
 
-        this.name = name;
-        if (bytes) {
-            this.bytes = name.getBytes(CharsetUtil.US_ASCII);
-        } else {
-            this.bytes = null;
-        }
+        this.name = new AsciiString(name);
+        nameAsString = name;
     }
 
     /**
-     * Returns the name of this getMethod.
+     * Returns the name of this method.
      */
-    public String name() {
+    public AsciiString name() {
```

Comment:
@Scottmitch This is an interesting question. I think a user would feel less surprising to remove a deprecated method instead of changing its return type and un-deprecating it.

How about adding `asciiName()` and also keep `name()`? `HttpMethod` is very similar to `Enum`, so it would be less surprising when the return type of `name()` remains as `String`.